src/core/structure.js

```diff
@@ -68,7 +68,9 @@ function scanSections(sections, maxTocLevel, { prefix = "" } = {}) {
       : appendixMode
       ? alphabet.charAt(index - lastNonAppendix)
       : prefix + index;
-    const level = Math.ceil(secno.length / 2);
+    // Thijs Brentjens, Geonovum: fix for determining the level:
+    // count the number of dots in the section number, to determine the level
+    const level = (secno.match(/\./g)||[]).length + 1;
```

Comment:
Maybe use this?
```suggestion
    const level = secno.split(".").length;
```

Reply:
Hmm, I clicked on something accidentally. Your change is simpler, better. I'll use that